lib/orderbook/OrderBook.ts

```diff
@@ -29,102 +39,153 @@ class OrderBook {
 
-  public async init() {
-    const [pairs, peerOrders, ownOrders] = await Promise.all([
-      this.repository.getPairs(),
-      this.repository.getPeerOrders(),
-      this.repository.getOwnOrders(),
-    ]);
-
-    const peerBuyOrdersByPairs = groupBy(peerOrders.buyOrders, order => order.pairId);
-    const peerSellOrdersByPairs = groupBy(peerOrders.sellOrders, order => order.pairId);
-    const ownBuyOrdersByPairs = groupBy(ownOrders.buyOrders, order => order.pairId);
-    const ownSellOrdersByPairs = groupBy(ownOrders.sellOrders, order => order.pairId);
+  public init = async () => {
+    const pairs = await this.repository.getPairs();
 
     pairs.forEach((pair) => {
       this.matchingEngines[pair.id] = new MatchingEngine(
         pair.id,
         this.config.internalmatching,
-        peerBuyOrdersByPairs[pair.id],
-        peerSellOrdersByPairs[pair.id],
-        ownBuyOrdersByPairs[pair.id],
-        ownSellOrdersByPairs[pair.id],
       );
+      this.ownOrders[pair.id] = this.initOrders();
+      this.peerOrders[pair.id] = this.initOrders();
     });
 
     this.pairs = pairs;
   }
 
+  private initOrders = (): Orders => {
+    return {
+      buyOrders: [],
+      sellOrders: [],
+    };
+  }
+
   /**
    * Returns the list of available trading pairs.
    */
-  public getPairs(): Promise<db.PairInstance[]> {
+  public getPairs = (): Promise<db.PairInstance[]> => {
     return this.repository.getPairs();
   }
 
   /**
    * Returns lists of buy and sell orders of peers sorted by price.
    */
-  public getPeerOrders(pairId: string, maxResults: number): Promise<Orders> {
-    return this.repository.getPeerOrders(pairId, maxResults);
+  public getPeerOrders = (pairId: string, maxResults: number): Orders => {
+    return this.getOrders(maxResults, this.peerOrders[pairId]);
   }
 
   /*
   * Returns lists of the node's own buy and sell orders sorted by price.
   */
-  public getOwnOrders(pairId: string, maxResults: number): Promise<Orders> {
-    return this.repository.getOwnOrders(pairId, maxResults);
+  public getOwnOrders = (pairId: string, maxResults: number): Orders => {
+    return this.getOrders(maxResults, this.ownOrders[pairId]);
+  }
+
+  private getOrders = (maxResults: number, orders: Orders) => {
+    this.sortOrdersByPrice(orders);
+    if (maxResults > 0) {
+      const result = this.initOrders();
+
+      let index = 0;
+      while (index < maxResults) {
+        if (orders.buyOrders.length > index) {
+          result.buyOrders.push(orders.buyOrders[index]);
+        }
+        if (orders.sellOrders.length > index) {
+          result.sellOrders.push(orders.sellOrders[index]);
+        }
+        index = index + 1;
+      }
+
+      return result;
+    } else {
+      return orders;
+    }
   }
 
-  public addLimitOrder = async (order: orders.OwnOrder): Promise<matchingEngine.MatchingResult>  => {
+  // TODO: sort when adding new order or getting existing ones?
+  private sortOrdersByPrice = (orders: Orders) => {
```

Comment:
I don't think it make sense, performance-wise, to sort the orders on each get request. That's why we use the priority queues in the matching engine, so the list will be sorted each time a new order is added (slow writes, fast reads). 
It's not clear if we even need this result sorted. For the `GETORDERS` response, the peer will need to re-sort anyway, since he can't rely on it to already be sorted. And for the `getInfo` response, i'm not sure we need it sorted, and even if so, we can support that separately.
At the moment I guess you can just use the priority queue from MatchingEngine, if you want it sorted. But since we're going to merge the own/peer orders queues (#196) it won't be usable anymore soon, because you wouldn't be able to get the sorted `ownOrders` without the `peerOrders`. So I think that in the meanwhile you can just return it unsorted.